### ISSUE TITLE
[Bug]: Lint.sh from examples can fail on long-lived CI workers

### ISSUE THREAD

**Issue by mortenmj** [aspect-build/rules_lint#45]
### What happened?

If I make a PR adding some new component to our repository, this will cause a linting report to be written to bazel-out. If the same agent later builds some other branch, where this new component is not present in the source tree, the report will still be present in bazel-out. If the PR has a linting failure, this will also cause other branches to fail.

### Version

Development (host) and target OS/architectures:

Output of `bazel --version`:

bazel 6.4.0

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file:

bazel_dep(name = "aspect_bazel_lib", version = "2.0.0-rc0")
bazel_dep(name = "aspect_rules_lint", version = "0.0.0")

git_override(
    module_name = "aspect_rules_lint",
    commit = "699cd2c9719bfb3bfc3a9f8c6a7006f0a058c067",
    remote = "https://github.com/aspect-build/rules_lint.git",
)

Language(s) and/or frameworks involved:

Python

### How to reproduce

```shell
Create a branch which adds some file with a linting failure. Run the /examples/lint.sh on this branch. Then run on some other branch which doesn't have the same file.
```

**Comment by alexeagle** [comment]
Yes, this is a known issue with the `lint.sh` script - it cannot listen on the Build Event Stream so it doesn't know which lint files are newly written.

I recommend running `bazel lint` instead, by adding the Aspect CLI (can do this with just a couple environment variables on CI)
Would that work for you @mortenmj ?

**Comment by mortenmj** [comment]
We're not currently using the Aspect CLI. Currently we're running a modified version of lint.sh that reads writes a build event JSON file and gets the relevant reports from that. Happy to share it if you're interested.

**Comment by alexeagle** [comment]
Yes, I'm definitely interested, if we can make that simple enough we could build it into the shell version?

**Comment by mortenmj** [comment]
It's pretty simple. Feel free to use it if you'd like.

```
#!/usr/bin/env bash
#
set -euo pipefail

if [ "$#" -eq 0 ]; then
    echo "usage: lint.sh [target pattern...]"
    exit 1
fi

config=(--config=lint)

# File to store the build event output in
logfile=/tmp/aspect_rules_lint.json

# Produce report files
bazel build "${config[@]}" --build_event_json_file="$logfile" "$@"

valid_reports=$(jq -r '.namedSetOfFiles | values | .files[] | ((.pathPrefix | join("/")) + "/" + .name)' "$logfile")

exit_code=0

# Show the results.
while IFS= read -r report; do
    # Exclude coverage reports, and check if the report is empty.
    if [[ "$report" == *coverage.dat ]] || [[ ! -s "$report" ]]; then
        # Report is empty. No linting errors.
        continue
    fi
    echo "From ${report}:"
    cat "${report}"
    echo

    exit_code=1
done <<<"$valid_reports"

exit $exit_code
```

**Comment by mortenmj** [comment]
Forgot to add, we set these flags:

```
build:lint --aspects //build/aspects:lint.bzl%mypy,//build/aspects:lint.bzl%ruff,//build/aspects:lint.bzl%shellcheck
build:lint --output_groups=rules_lint_report
build:lint --remote_download_regex=".*aspect_rules_lint.report"
build:lint --remote_download_toplevel  # unsure if this one is needed
```
